packages/lodestar/src/network/gossip/validation/aggregateAndProof.ts

```diff
@@ -0,0 +1,170 @@
+import {AggregateAndProof, Attestation, SignedAggregateAndProof, Slot} from "@chainsafe/lodestar-types";
+import {ExtendedValidatorResult} from "../constants";
+import {toHexString} from "@chainsafe/ssz";
+import {ATTESTATION_PROPAGATION_SLOT_RANGE, MAXIMUM_GOSSIP_CLOCK_DISPARITY} from "../../../constants";
+import {
+  computeEpochAtSlot,
+  computeStartSlotAtEpoch,
+  getCurrentSlot,
+  isAggregatorFromCommitteeLength,
+} from "@chainsafe/lodestar-beacon-state-transition";
+import {isAttestingToInValidBlock} from "./attestation";
+import {getAttestationPreState} from "../utils";
+import {processSlots} from "@chainsafe/lodestar-beacon-state-transition/lib/fast/slot";
+import {Signature} from "@chainsafe/bls";
+import {isValidIndexedAttestation} from "@chainsafe/lodestar-beacon-state-transition/lib/fast/block/isValidIndexedAttestation";
+import {isValidAggregateAndProofSignature, isValidSelectionProofSignature} from "./utils";
+
+export async function validateGossipAggregateAndProof(
+  config: IBeaconConfig,
+  chain: IBeaconChain,
+  db: IBeaconDb,
+  logger: ILogger,
+  signedAggregateAndProof: SignedAggregateAndProof
+): Promise<ExtendedValidatorResult> {
+  logger.profile("gossipAggregateAndProofValidation");
+  const aggregateAndProof = signedAggregateAndProof.message;
+  const aggregate = aggregateAndProof.aggregate;
+  const root = config.types.AggregateAndProof.hashTreeRoot(aggregateAndProof);
+  const attestationRoot = config.types.Attestation.hashTreeRoot(aggregate);
+  const logContext = getLogContext(aggregate, aggregateAndProof, root, attestationRoot);
+  logger.verbose("Started gossip aggregate and proof validation", logContext);
+  if (!isValidAggregateAndProofSlot(config, chain.getGenesisTime(), aggregate.data.slot)) {
+    logger.warn("Ignored gossip aggregate and proof", {
+      reason: "invalid slot time",
+      currentSlot: getCurrentSlot(config, chain.getGenesisTime()),
+      ...logContext,
+    });
+    //TODO: aggregate and proof pool to wait until proper slot to replay
+    return ExtendedValidatorResult.ignore;
+  }
+  if (await db.seenAttestationCache.hasAggregateAndProof(aggregateAndProof)) {
+    logger.warn("Ignored gossip aggregate and proof", {
+      reason: "seen aggregator and target index",
+      targetEpoch: aggregate.data.target.epoch,
+      ...logContext,
+    });
+    return ExtendedValidatorResult.ignore;
+  }
+  if (!hasAttestationParticipants(aggregate)) {
+    logger.warn("Rejected gossip aggregate and proof", {
+      reason: "missing attestation participants",
+      ...logContext,
+    });
+    return ExtendedValidatorResult.reject;
+  }
+
+  if (await isAttestingToInValidBlock(db, aggregate)) {
+    logger.warn("Rejected gossip aggregate and proof", {
+      reason: "attesting to invalid block",
+      ...logContext,
+    });
+    return ExtendedValidatorResult.reject;
+  }
+
+  //TODO: check pool of aggregates if already seen (not a dos vector check)
```

Comment:
is this something that will be done in this PR?  or does a separate issue need to be made?

Reply:
Separate issue, not really a priority i guess